src/FluentMigrator.Runner.Core/Conventions/DefaultPrimaryKeyNameConvention.cs

```diff
@@ -29,23 +32,24 @@ public class DefaultPrimaryKeyNameConvention : IColumnsConvention
         /// <inheritdoc />
         public IColumnsExpression Apply(IColumnsExpression expression)
         {
+            if (expression.Columns.Count(x => x.IsPrimaryKey) > 1)
+                throw new InvalidOperationException("Error creating table with multiple primary keys.");
+
             foreach (var columnDefinition in expression.Columns)
             {
+                if (string.IsNullOrEmpty(columnDefinition.Name))
+                    throw new ArgumentException("An object or column name is missing or empty.");
+
                 if (columnDefinition.IsPrimaryKey && string.IsNullOrEmpty(columnDefinition.PrimaryKeyName))
                 {
                     var tableName = string.IsNullOrEmpty(columnDefinition.TableName)
                         ? expression.TableName
                         : columnDefinition.TableName;
-                    columnDefinition.PrimaryKeyName = GetPrimaryKeyName(tableName);
+                    columnDefinition.PrimaryKeyName = $"PK_{tableName}_{columnDefinition.Name}";
```

Comment:
I think the current behavior is what most users prefer. The actual column name(s) don't really matter to most developers except when needing to change the data type, as that requires dropping and recreating the column name.

Additionally, most DB engines like SQL Server have limit of, say, 255 characters for system object names.

Awhile ago I had the idea of an Opinions assembly that has a different set of default conventions and even some syntax shortcuts. This could live in something like that.

Reply:
@jzabroski 

If the column name doesn't matter then why shouldn't it be deterministic?

If the name is too long the user will just get an exception, as in the case of `DefaultConstraintNameConvention`.

I don't think this is a good enough reason to complicate the project's ecosystem.

Reply:
From reading the original issue, it sounds like Create.PrimaryKey() uses DefaultConstraintNameConvention - this is a bug. It should use DefauktPrimaryKeyNameConvention.

Default constraints should just be for default values specified by the ANSI SQL DDL `DEFAULT` keyword.